### ISSUE TITLE
PHPCS reports invalid indenting of closure body (probably due to return type hint definition)

### ISSUE THREAD

**Issue by Ocramius** [squizlabs/PHP_CodeSniffer#876]
Taking following snippet:

```php
    /**
     * Retrieves all abstract methods to be proxied
     *
     * @param ReflectionClass $originalClass
     *
     * @return MethodGenerator[]
     */
    private function getAbstractProxiedMethods(ReflectionClass $originalClass) : array
    {
        return array_map(
            function (ReflectionMethod $method) : ProxyManagerMethodGenerator {
                return ProxyManagerMethodGenerator::fromReflection(
                    new MethodReflection($method->getDeclaringClass()->getName(), $method->getName())
                )->setAbstract(false);
            },
            ProxiedMethodsFilter::getAbstractProxiedMethods($originalClass)
        );
    }
```

PHPCS reports:

```
FILE: ...er/src/ProxyManager/ProxyGenerator/LazyLoadingGhostGenerator.php
----------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
----------------------------------------------------------------------
 148 | ERROR | [x] Multi-line function call not indented correctly;
     |       |     expected 12 spaces but found 16
----------------------------------------------------------------------
PHPCBF CAN FIX THE 1 MARKED SNIFF VIOLATIONS AUTOMATICALLY
----------------------------------------------------------------------
```

If I change the code to following (incorrect, of course):

```php
    /**
     * Retrieves all abstract methods to be proxied
     *
     * @param ReflectionClass $originalClass
     *
     * @return MethodGenerator[]
     */
    private function getAbstractProxiedMethods(ReflectionClass $originalClass) : array
    {
        return array_map(
            function (ReflectionMethod $method) : ProxyManagerMethodGenerator {
            return ProxyManagerMethodGenerator::fromReflection(
                new MethodReflection($method->getDeclaringClass()->getName(), $method->getName())
            )->setAbstract(false);
            },
            ProxiedMethodsFilter::getAbstractProxiedMethods($originalClass)
        );
    }
```

Then PHPCS reports:

```
FILE: ...er/src/ProxyManager/ProxyGenerator/LazyLoadingGhostGenerator.php
----------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
----------------------------------------------------------------------
 148 | ERROR | [x] Line indented incorrectly; expected at least 16
     |       |     spaces, found 12
----------------------------------------------------------------------
PHPCBF CAN FIX THE 1 MARKED SNIFF VIOLATIONS AUTOMATICALLY
----------------------------------------------------------------------
```

**Comment by gsherwood** [comment]
I tried that snippet (and found the file on github and tested that) and didn't get any errors in the latest stable version. But I did get errors on 2.5.0. I think this might be the same issue as bug #835, which was fixed and released in 2.5.1.

Can you upgrade to 2.5.1 and test again please?

**Comment by Ocramius** [comment]
@goatherd I'll try against master while at phpbnl, thanks :-)

**Comment by gsherwood** [comment]
Closing as I'm pretty sure this is fixed. Please comment if not.

**Comment by Ocramius** [comment]
Thanks, sorry for not responding before - didn't manage to try it in isolation  :-\
